src/main/java/edu/ucsb/cs156/rec/repositories/RecommendationRequestRepository.java

```diff
@@ -0,0 +1,28 @@
+package edu.ucsb.cs156.rec.repositories;
+
+import edu.ucsb.cs156.rec.entities.RecommendationRequest;
+
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import java.util.Optional;
+
+/**
+ * The RecommendationRequestRepository is a repository for RecommendationRequest entities.
+ */
+
+@Repository
+public interface RecommendationRequestRepository extends CrudRepository<RecommendationRequest, Long> {
+  /**
+   * This method returns all RecommendationRequest entities with a given requesterName.
+   * @param requesterName name of the person making the reqest 
+   * @return all RecommendationRequest entities with a given requesterName
+   */
+  Optional<RecommendationRequest> findByRequesterName(String requesterName);
```

Comment:
This will definitely not work. Do you know how many `David Wang` student I've worked with?   They will all see each other's requests, which is not good.

You need to guarantee uniqueness.   A userId field does that.

Please replace with:

```suggestion
  Optional<RecommendationRequest> findByRequesterId(Long requesterId);
```

Reply:
Fixed